parl/remote/remote_decorator.py

```diff
@@ -66,36 +66,23 @@ def _create_reply_socket(self, remote_ip, remote_port):
             and waits for requests (e.g. call a function) from server side.
             """
             if remote_ip is None:
-                client_ip = get_ip_address()
-            else:
-                client_ip = remote_ip
+                remote_ip = get_ip_address()
 
             self.zmq_context = zmq.Context()
             socket = self.zmq_context.socket(zmq.REP)
 
-            free_port = None
             if remote_port is None:
-                for port in range(6000, 8000):
-                    try:
-                        socket.bind("tcp://*:{}".format(port))
-                        logger.info(
-                            "[_create_reply_socket] free_port:{}".format(port))
-                        free_port = port
-                        break
-                    except zmq.error.ZMQError:
-                        logger.warning(
-                            "[_create_reply_socket]cannot bind port:{}, retry".
-                            format(port))
+                try:
+                    remote_port = socket.bind_to_random_port(addr="tcp://*")
+                except zmq.ZMQBindError:
+                    logger.error(
+                        'Can not bind to random port, please set remote_port manually.'
```

Comment:
'Can not bind to a random port, please set remote_port manually.

Reply:
done